tests/components/knx/test_config_flow.py

```diff
@@ -908,6 +1070,7 @@ async def test_options_flow_connection_type(
     gateway = _gateway_descriptor("192.168.0.1", 3675)
 
     await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    hass.data[DOMAIN] = Mock()  # GatewayScanner uses running XKNX() in options flow
```

Comment:
If we need the integration setup, set up the integration without patching the integration setup function. Just patch the library. Never set things in hass.data in tests.

Reply:
Addressed in #83208